src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -46,17 +49,20 @@ internal abstract class BrokerClient : IBrokerClient {
 
         public string Name { get; }
         public Uri Uri { get; }
+        public string RCommandLineArguments { get; }
         public bool IsRemote => !Uri.IsFile;
         public bool IsVerified { get; protected set; }
 
-        protected BrokerClient(string name, Uri brokerUri, string interpreterId, ICredentialsDecorator credentials, IActionLog log, IConsole console) {
+        protected BrokerClient(string name, Uri brokerUri, string rCommandLineArguments, string interpreterId, ICredentialsDecorator credentials, IActionLog log, IConsole console) {
             Name = name;
             Uri = brokerUri;
             Log = log;
 
+            RCommandLineArguments = rCommandLineArguments?.Trim() ?? string.Empty;
             _interpreterId = interpreterId;
             _credentials = credentials;
             _console = console;
+            _parametersHash = CalculateAdditionalParametersHash();
```

Comment:
`_parametersHash` is required to distinguish different connections from the same user to the same machine for different parameters (command line arguments, interpreter, etc.).